docs/API.md

```diff
@@ -0,0 +1,176 @@
+# Shipyard API
+
+Logically, the API has three parts to handle the three areas of functionality in Shipyard.
+
+1. Document Staging
+2. Action Handling 
+3. Airflow Monitoring 
+
+There are several standard error responses that should be handled appropriately by a client of this API:
+* 401 Authentication Error  
+If not authenticated
+* 403 Forbidden  
+If not authorized
+* 404 Not found  
+If a variable in the path doesn't reference a valid resource 
+* 503 Service Unavailable  
+Error if an upstream system cannot be reached (e.g. Deckhand for documents, Airflow for actions)
+
+---
+## <a name="DocumentStagingAPI"></a> Document Staging API
+Shipyard will serve as the entrypoint for documents and secrets into a site. At any point in time, there will be two represented versions of documents in a site that are accessible via this API. 
+
+* The lastDeployed version, which represents the last version of documents that successfully completed deployment.
+* The staged version, which represents the document set that has been ingested by this API since the lastDeployed version.
+
+### /v1.0/configDocs  
+Represents the site configuration documents.  
+
+#### POST 
+Ingests site configuration documents. Synchronous.
+##### Responses
+* 200 OK  
+If the documents are successfully ingested, even with validation failures. Response message includes:
```

Comment:
Are we returning the GET as a Header like some REST API and not in the Payload perhaps?
Any idea what this validation results looks like ? If its useful for an application upstream which you define a proper model for application consumption.